### ISSUE TITLE
printlns interfere with show during first time setup

### ISSUE THREAD

**Issue by ghostbuster91** [com-lihaoyi/mill#2811]
Hi,

When calling `./mill show version` for the very first time on a new machine mill will print on stdout:
```
Preparing Java 17.0.8.1 runtime; this may take a minute or two ...
"0.1.1"
```

## Expected behavior
Stdout should only contain the result of the `show` command nothing besides that. 

## Actual behavior
Stdout contains information about preparing java together with results of the `show` command.

## Reproduction steps

Use following docker file:
```
FROM ubuntu:22.04

# Install OpenJDK-8
RUN apt-get update && \
    apt-get install -y openjdk-17-jdk curl jq && \
    apt-get clean;

COPY . /src/
WORKDIR /src
RUN ./mill show version > output
RUN cat output
```

mill-version: 0.11.4

The offending line is here https://github.com/com-lihaoyi/mill/blob/7c2114e3895bac6f60e641e854906b2139e7ec84/runner/src/mill/runner/MillMain.scala#L193

Somehow related to #2680

**Comment by lefou** [comment]
Looks reasonable. Instead, we should use the `logger` in scope, as it`s using output streams that get properly redirected by `show`.

**Comment by kubukoz** [comment]
@lefou any idea why it wasn't ultimately solved by #2689 ?

**Comment by lefou** [comment]
I wonder whether both lines come from STDOUT at all? `mill show` redirects all other output to `STDERR`, but if you don't pipe or redirect it, both output streams will still appear next to each other in the terminal.

**Comment by lihaoyi** [comment]
The setting up of the Java runtime happens pretty early; it might possibly happen before we set up the stream redirection

**Comment by lefou** [comment]
> @lefou any idea why it wasn't ultimately solved by #2689 ?

No, not really. The code location you linked is already inside a `SystemStreams.withStreams` block, so any configured redirections to `System.{in,out,err}` and `Console.{in,out,err}` should be in place. Some debugging might be necessary to better understand what's going on.
